web/src/site-admin/SiteAdminExportConfigPage.tsx

```diff
@@ -0,0 +1,146 @@
+import gitoliteSchemaJSON from '../../../schema/gitolite.schema.json'
+import otherExternalServiceSchemaJSON from '../../../schema/other_external_service.schema.json'
+import phabricatorSchemaJSON from '../../../schema/phabricator.schema.json'
+import settingsSchemaJSON from '../../../schema/settings.schema.json'
+import siteSchemaJSON from '../../../schema/site.schema.json'
+import { PageTitle } from '../components/PageTitle'
+import { ExternalServiceKind } from '../../../shared/src/graphql/schema'
+
+/**
+ * Minimal shape of a JSON Schema. These values are treated as opaque, so more specific types are
+ * not needed.
+ */
+interface JSONSchema {
+    $id: string
+    definitions?: Record<string, { type: string }>
+}
+
+const externalServices: [ExternalServiceKind, JSONSchema][] = [
+    [ExternalServiceKind.AWSCODECOMMIT, awsCodeCommitJSON],
+    [ExternalServiceKind.BITBUCKETCLOUD, bitbucketCloudSchemaJSON],
+    [ExternalServiceKind.BITBUCKETSERVER, bitbucketServerSchemaJSON],
+    [ExternalServiceKind.GITHUB, githubSchemaJSON],
+    [ExternalServiceKind.GITLAB, gitlabSchemaJSON],
+    [ExternalServiceKind.GITOLITE, gitoliteSchemaJSON],
+    [ExternalServiceKind.OTHER, otherExternalServiceSchemaJSON],
+    [ExternalServiceKind.PHABRICATOR, phabricatorSchemaJSON],
+]
+
+const allConfigSchema = {
+    $id: 'all.schema.json#',
+    allowComments: true,
+    additionalProperties: false,
+    properties: {
+        site: siteSchemaJSON,
+        externalServices: {
+            type: 'object',
+            properties: externalServices.reduce<
+                Partial<{ [k in ExternalServiceKind]: { type: string; items: JSONSchema } }>
+            >((properties, [kind, schema]) => {
+                properties[kind] = {
+                    type: 'array',
+                    items: schema,
+                }
+                return properties
+            }, {}),
+        },
+        settings: {
+            type: 'object',
+            properties: {
+                subjects: {
+                    type: 'array',
+                    items: {
+                        type: 'object',
+                        properties: {
+                            __typename: {
+                                type: 'string',
+                            },
+                            settingsURL: {
+                                type: ['string', 'null'],
+                            },
+                            contents: {
+                                ...settingsSchemaJSON,
+                                type: ['object', 'null'],
+                            },
+                        },
+                    },
+                },
+                final: settingsSchemaJSON,
+            },
+        },
+    },
+    definitions: externalServices
+        .map(([, schema]) => schema.definitions)
+        .concat([siteSchemaJSON.definitions, settingsSchemaJSON.definitions])
+        .reduce(
+            (definitions, theseDefinitions) =>
```

Comment:
I find this terser version more readable. What do you think?

```suggestion
            (allDefinitions, definitions) => ({ ...(definitions || {}), ...allDefinitions })
```

Reply:
AFAIK the spread operator [should ignore](https://codesandbox.io/s/tender-hooks-qvvh5) falsy values, so it could just be
```ts
(allDefinitions, definitions) => ({ ...allDefinitions, ...definitions })
```